glfs.c

```diff
@@ -836,6 +874,56 @@ static int tcmu_glfs_handle_cmd(struct tcmu_device *dev, struct tcmulib_cmd *cmd
 	return ret;
 }
 
+#if GFAPI_VERSION770
+static int tcmu_glfs_to_sts(int rc)
+{
+	switch (rc) {
+	case 0:
+		return TCMU_STS_OK;
+	case -ENOTCONN:
+		return TCMU_STS_FENCED;
+	default:
+		return TCMU_STS_HW_ERR;
+	}
+}
+
+static int tcmu_glfs_lock(struct tcmu_device *dev, uint16_t tag)
+{
+	struct glfs_state *state = tcmur_dev_get_private(dev);
+	struct flock lock;
+	int ret;
+
+	lock.l_type = F_WRLCK | F_RDLCK;
+	lock.l_whence = SEEK_SET;
+	lock.l_start = 0;
+	lock.l_len = 0;
+
+	ret = glfs_file_lock(state->gfd, F_SETLK, &lock, GLFS_LK_MANDATORY);
```

Comment:
If you already have the lock, then call this again will it return success or failure?

Reply:
If the lock is already took by any of the other nodes, it will try to break it and take it in local node.

In the local node, all the cases are the same with the linux mandatory lock. Here it will be blocked if any other process has took it.

Reply:
For this the test is not finished yet due to there is one bug in the gluster backend, which is till not fixed yet by the gluster core team.